pkg/sql/opt/norm/testdata/rules/decorrelate

```diff
@@ -8132,3 +8132,646 @@ project
+ │    ├── columns: x:8!null y:9!null
+ │    ├── key: (8)
+ │    ├── fd: ()-->(9)
+ │    ├── scan xy
+ │    │    ├── columns: x:8!null y:9
+ │    │    ├── key: (8)
+ │    │    └── fd: (8)-->(9)
+ │    └── filters
+ │         └── y:9 = 1000 [outer=(9), constraints=(/9: [/1000 - /1000]; tight), fd=()-->(9)]
+ └── filters
+      └── x:8 = k:1 [outer=(1,8), constraints=(/1: (/NULL - ]; /8: (/NULL - ]), fd=(1)==(8), (8)==(1)]
+
+# No-op because all filters reference xy.
+norm expect-not=HoistUnboundFilterFromExistsSubquery
+SELECT * FROM a WHERE EXISTS (
+  SELECT * FROM xy WHERE xy.x = a.k AND xy.y + a.i = 1000
+);
+----
+semi-join (hash)
+ ├── columns: k:1!null i:2 f:3 s:4 j:5
+ ├── immutable
+ ├── key: (1)
+ ├── fd: (1)-->(2-5)
+ ├── scan a
+ │    ├── columns: k:1!null i:2 f:3 s:4 j:5
+ │    ├── key: (1)
+ │    └── fd: (1)-->(2-5)
+ ├── scan xy
+ │    ├── columns: x:8!null y:9
+ │    ├── key: (8)
+ │    └── fd: (8)-->(9)
+ └── filters
+      ├── x:8 = k:1 [outer=(1,8), constraints=(/1: (/NULL - ]; /8: (/NULL - ]), fd=(1)==(8), (8)==(1)]
+      └── (y:9 + i:2) = 1000 [outer=(2,9), immutable]
+
+# No-op case with NOT EXISTS.
+norm expect-not=HoistUnboundFilterFromExistsSubquery
+SELECT * FROM a WHERE NOT EXISTS (
+  SELECT * FROM xy WHERE xy.x = a.k AND a.i = 1000
+);
+----
+anti-join (hash)
+ ├── columns: k:1!null i:2 f:3 s:4 j:5
+ ├── key: (1)
+ ├── fd: (1)-->(2-5)
+ ├── scan a
+ │    ├── columns: k:1!null i:2 f:3 s:4 j:5
+ │    ├── key: (1)
+ │    └── fd: (1)-->(2-5)
+ ├── scan xy
+ │    ├── columns: x:8!null
+ │    └── key: (8)
+ └── filters
+      ├── x:8 = k:1 [outer=(1,8), constraints=(/1: (/NULL - ]; /8: (/NULL - ]), fd=(1)==(8), (8)==(1)]
+      └── i:2 = 1000 [outer=(2), constraints=(/2: [/1000 - /1000]; tight), fd=()-->(2)]
+
+# No-op because the top expression in the subquery isn't a Select.
+norm expect-not=HoistUnboundFilterFromExistsSubquery
+SELECT * FROM a WHERE EXISTS (
+  SELECT sum(xy.y) OVER () FROM xy WHERE xy.x = a.k AND a.i = 1000
+);
+----
+project
+ ├── columns: k:1!null i:2!null f:3 s:4 j:5
+ ├── key: (1)
+ ├── fd: ()-->(2), (1)-->(3-5)
+ └── inner-join (hash)
+      ├── columns: k:1!null i:2!null f:3 s:4 j:5 x:8!null
+      ├── multiplicity: left-rows(zero-or-one), right-rows(zero-or-one)
+      ├── key: (8)
+      ├── fd: ()-->(2), (1)-->(3-5), (1)==(8), (8)==(1)
+      ├── select
+      │    ├── columns: k:1!null i:2!null f:3 s:4 j:5
+      │    ├── key: (1)
+      │    ├── fd: ()-->(2), (1)-->(3-5)
+      │    ├── scan a
+      │    │    ├── columns: k:1!null i:2 f:3 s:4 j:5
+      │    │    ├── key: (1)
+      │    │    └── fd: (1)-->(2-5)
+      │    └── filters
+      │         └── i:2 = 1000 [outer=(2), constraints=(/2: [/1000 - /1000]; tight), fd=()-->(2)]
+      ├── scan xy
+      │    ├── columns: x:8!null
+      │    └── key: (8)
+      └── filters
+           └── x:8 = k:1 [outer=(1,8), constraints=(/1: (/NULL - ]; /8: (/NULL - ]), fd=(1)==(8), (8)==(1)]
+
+norm expect=HoistUnboundJoinFilterFromExistsSubquery
```

Comment:
nit: Is this test supposed to be in the section below?

Reply:
Yes, Done.